src/main/assets/js/courts-table-search.ts

```diff
@@ -0,0 +1,249 @@
+        false, orderToggleState.ASC, orderToggleState.INACTIVE)
+      : CourtsTableSearch.setUpTableData(
+        $(this.searchCourtsFilterId).val() as string,
+        $(`#main-content input[name=${this.toggleClosedCourtsDisplay}]`).prop('checked'),
+        $(this.courtsNameAscToggleId).val() as string,
+        $(this.courtsUpdatedAscToggleId).val() as string);
+
+    filterName == 'name' ?
+      CourtsTableSearch.switchTableClasses($(this.courtsNameAscToggleId),
+        $(this.tableCourtsNameId), $(this.tableCourtsUpdatedId)) :
+      CourtsTableSearch.switchTableClasses($(this.courtsUpdatedAscToggleId),
+        $(this.tableCourtsUpdatedId), $(this.tableCourtsNameId));
+  }
+
+  /**
+   *
+   * Retrieves the filtered results from the toggles and search, and sets the html
+   * of the tbody element of the table accordingly
+   *
+   * @param searchFilterValue the search value from the 'search courts or tribunals' input field
+   * @param includeClosedCourts whether or not to include closed courts in the result
+   * @param orderNameAscendingFilter whether to sort by name asc or desc
+   * @param orderUpdatedAscendingFilter whether to sort by last updated date asc or desc
+   * @private
+   */
+  private static setUpTableData(searchFilterValue: string, includeClosedCourts: boolean,
+    orderNameAscendingFilter: string, orderUpdatedAscendingFilter: string): void {
+    const filteredCourts = CourtsTableSearch.filterCourts(this.getExistingTableData(),
+      searchFilterValue, includeClosedCourts,
+      orderNameAscendingFilter, orderUpdatedAscendingFilter);
+    $(this.courtsResultsSection).html(CourtsTableSearch.getCourtsTableBody(filteredCourts));
+    searchFilterValue.length
+      ? $(this.numberOfCourts).show().text('Showing '
+        + filteredCourts.filter(d => d.visible).length + ' results')
+      : $(this.numberOfCourts).hide();
+  }
+
+  /**
+   *
+   * Get the court data by extracting it from the table
+   *
+   * @private
+   */
+  private static getExistingTableData(): CourtItem[] {
+    const courtItems = [] as CourtItem[];
+    $(this.tableData).find('tr').each(function(i, row) {
+      if (i == 0)
+        // the first index is for the column names; i.e name, displayed, updated_at
+        return;
+
+      const courtItem = {} as CourtItem;
+      $(row).find('td').each(function(i, dataCell) {
+
+        switch ($(dataCell).attr('type')) {
+          case 'name':
+            courtItem.name = $(dataCell).attr('value');
+            courtItem.slug = $(dataCell).attr('name');
+            break;
+          case 'displayed':
+            courtItem.displayed = $(dataCell).attr('value') === 'true';
+            break;
+          case 'updated_at':
+            courtItem.updatedAt = $(dataCell).attr('value');
+            break;
+          default:
+            break;
+        }
+      });
+
+      courtItem.row = $(row);
```

Comment:
Does this need to be a jquery type?

Reply:
I'm not too sure, is it more performant to do it in javascript? I am kind of keeping to using either mostly jquery or javascript, and it works using jquery for the moment